.clang-tidy

```diff
@@ -28,6 +28,7 @@ Checks:
     hicpp-signed-bitwise,
     misc-*,
     modernize-*,
+    -modernize-use-trailing-return-type,
```

Comment:
We need to consider whether this is important or not.

My immediate thought is, trailing return types are good for keeping your function names aligned. There are certainly reasons for using trailing return types in generic code sometimes. However, it's nothing wrong with leading return types. It's better to maintain a consistent style for each library/API IMO.

Will think about this more.